src/main.cpp

```diff
@@ -3314,9 +3314,15 @@ static bool ActivateBestChainStep(CValidationState& state, const CChainParams& c
                 if (!IsInitialBlockDownload()) {
                     // Notify external zmq listeners about the new tip.
                     GetMainSignals().UpdatedBlockTip(pindexConnect);
+                }
 
-                    // Update the UI after each block if we are close to finishing IBD
+                // Update the UI at least every 5 seconds just in case we get in a long loop
+                // as can happen during IBD.  We need an atomic here because there may be other
+                // threads running concurrently.
+                static atomic<int64_t> nLastUpdate = {GetTime()};
```

Comment:
I am concerned that the initializer itself might not be thread safe.  the highest upvoted comment here seems to support that compilers do not handle this consistently: http://stackoverflow.com/questions/4590626/c-static-variables-in-multithreaded-program.

would it be possible to use an actual global variable instead of a static?
The global could be initialized to zero and the only effect would be that the if statement evaluates to true the first time...